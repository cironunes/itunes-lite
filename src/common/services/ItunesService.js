const BASE_URL = 'https://itunes.apple.com';

export const searchAlbumByTerm = term =>
  window.fetch(`${BASE_URL}/search?entity=album&term=${term}`);

export const getAlbumById = id =>
  window.fetch(`${BASE_URL}/lookup?entity=song&id=${id}`);
