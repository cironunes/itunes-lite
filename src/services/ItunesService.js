export const searchAlbumByTerm = (term) => {
  return window.fetch(`https://itunes.apple.com/search?entity=album&term=${term}`);
};

export const getAlbumById = (id) => {
  return window.fetch(`https://itunes.apple.com/lookup?entity=song&id=${id}`)
}
