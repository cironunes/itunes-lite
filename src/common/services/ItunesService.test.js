import { searchAlbumByTerm } from './ItunesService';

const json = { json: () => ({ results: [1, 2, 3] }) };

fetch = jest
  .fn()
  .mockImplementation(() => new Promise(resolve => resolve(json)));

describe('Service: Itunes', () => {
  test('Search album by term', async () => {
    const response = await searchAlbumByTerm('jack');
    const result = await response.json();

    expect(result.results).toMatchSnapshot();
  });
});
