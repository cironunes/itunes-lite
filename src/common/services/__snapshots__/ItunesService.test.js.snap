// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Service: Itunes Search album by term 1`] = `
Array [
  1,
  2,
  3,
]
`;
